v20.2/use-cloud-storage-for-bulk-operations.md

```diff
@@ -75,12 +56,106 @@ URLs for [changefeeds](stream-data-out-of-cockroachdb-using-changefeeds.html) sh
+  - `specified`: Pass the JSON object for authentication to the `CREDENTIALS` parameter. The JSON key object needs to be base64-encoded (using the standard encoding in [RFC 4648](https://tools.ietf.org/html/rfc4648)).
+
+    To access the storage bucket with specified credentials, it's necessary to [create a service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) and add the service account address to the permissions on the specific storage bucket. [The JSON object for authentication](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console) can be downloaded, encoded, and then passed to the `CREDENTIALS` parameter.
+
+    ~~~sql
+    BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=specified&CREDENTIALS={encoded key}';
+    ~~~
+
+  - `implicit`: The instance can use [environment data](https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application) from the service account to access resources. This will provide implicit access to the storage bucket.
+
+    ~~~sql
+    BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=implicit';
+    ~~~
+
+  - If `AUTH` is not provided, use the key provided in the `cloudstorage.gs.default.key` [cluster setting](cluster-settings.html). Otherwise, use environment data.
+
+    {{site.data.alerts.callout_info}}
+    **Deprecation notice:** Currently, GCS connections default to the `cloudstorage.gs.default.key` [cluster setting](cluster-settings.html). This default behavior will no longer be supported in v21.2. If you are relying on this default behavior, we recommend adjusting your queries and scripts to now specify the `AUTH` parameter you want to use. Similarly, if you are using the `cloudstorage.gs.default.key` cluster setting to authorize your GCS connection, we recommend switching to use `AUTH=specified` or `AUTH=implicit`. `AUTH=specified` will be the default behavior in v21.2 and beyond.
+    {{site.data.alerts.end}}
+
+### Azure Storage
+
+To access Azure storage containers, it is sometimes necessary to url encode the account key since it is base64-encoded and may contain `+`, `/`, `=` characters. For example:
+
+~~~sql
+BACKUP DATABASE <database> INTO 'azure://{container name}/{path}?AZURE_ACCOUNT_NAME={account name}&AZURE_ACCOUNT_KEY={url-encoded key}';
+~~~
+
+### HTTP
+
+If your environment requires an HTTP or HTTPS proxy server for outgoing connections, you can set the standard `HTTP_PROXY` and `HTTPS_PROXY` environment variables when starting CockroachDB. You can create your own HTTP server with [NGINX](use-a-local-file-server-for-bulk-operations.html). A custom root CA can be appended to the system's default CAs by setting the `cloudstorage.http.custom_ca` [cluster setting](cluster-settings.html), which will be used when verifying certificates from HTTPS URLs.
```

Comment:
it might be nice to link this mention of "environment variables" to our docs on them for background: https://www.cockroachlabs.com/docs/stable/cockroach-commands.html#environment-variables